portfolio/src/main/webapp/script.js

```diff
@@ -32,3 +32,15 @@ function gotoRandomSite() {
   // Change current location to new site.
   window.location.href = site;
 }
+
+function fetchGreeting() {
+    // Perform the fetch and store as promise
+    var greetingPromise = fetch('/data');
+
+    // Use callbacks on promise to extract the text and put into HTML element
+    greetingPromise.then(response => {
```

Comment:
I think response => response.text() is more idiomatic than response => {return response.text();}